### PR TITLE
Move most examples to use features instead of properties

### LINKED ISSUE #4035: examples/points_over_time.py is broken after #3730
## 🐛 Bug

`git bisect` points to #3730 as the commit that broke the example `points_over_time`. Something is off about the `face_color_mode` and how it's set/reset. It doesn't seem to stick, so as the layer gets updated the face colors do not change. @andy-sweet , any quick idea on where to look for the issue?

## To Reproduce

Run the example and move the dims slider; nothing changes, whereas it should update point colors.

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

 - Any other relevant information:

### PR DESCRIPTION
**Pull request by jni**
Fixes #4035

The only example I haven't changed is bbox_annotator. That uses current_properties and default_properties and I didn't know off the top of my head how to port those to the new features API. I'll let @andy-sweet deal with that one in a separate PR.

I think this PR contains only uncontroversial changes.

### CODE REVIEW

**Review by alisterburt — APPROVED** [review]
very nice! 

napari-subboxer uses the `current_properties` heavily and I need to port it to features - could you cc me if you do update that example @andy-sweet ?

File: examples/add_shapes_with_text.py [hunk 1]
```diff
@@ -19,14 +19,14 @@
...
-properties = {
+# create features
+features = {
     'likelihood': [21.23423, 51.2315, 100],
     'class': ['hand', 'face', 'camera'],
 }
 edge_color_cycle = ['blue', 'magenta', 'green']
 
-text_properties = {
+text_features = {
```
Comment (andy-sweet) [inline examples/add_shapes_with_text.py]: Maybe just call this `text`? These are effectively the parameters that will be passed to `TextManager`, so I think `text_parameters` could also work if that's too terse/unclear.

I think `text_features` confuses this with `Layer.features`.

File: examples/nD_points_with_properties.py [hunk 2]
```diff
@@ -19,26 +19,26 @@
 for z in range(blobs.shape[0]):
     points += [[z, 25, 25], [z, 25, 75], [z, 75, 25], [z, 75, 75]]
 
-# create the property for setting the face and edge color.
-face_property = np.array(
+# create the features for setting the face and edge color.
```
Comment (andy-sweet) [inline examples/nD_points_with_properties.py:22]: Rename this file to `nD_points_with_features.py`?

File: examples/show_points_based_on_property.py [hunk 3]
```diff
@@ -5,13 +5,15 @@
 from magicgui import magicgui
 
 
-# create points with a randomized "confidence" property
+# create points with a randomized "confidence" feature
```
Comment (andy-sweet) [inline examples/show_points_based_on_property.py:8]: Rename this file to `show_points_based_on_feature.py`?

File: examples/add_points_with_features.py [hunk 4]
```diff
@@ -46,13 +47,14 @@
 def toggle_point_annotation(viewer):
     selected_points = list(points_layer.selected_data)
     if len(selected_points) > 0:
-        good_point = points_layer.properties['good_point']
+        good_point = points_layer.features['good_point']
         good_point[list(selected_points)] = ~good_point[list(selected_points)]
```
Comment (andy-sweet) [inline examples/add_points_with_features.py]:
```suggestion
        good_point[selected_points] = ~good_point[selected_points]
```

Optional: not relevant these changes, so feel to ignore, but `selected_points` is already a `list` here.

**Review by andy-sweet — CHANGES_REQUESTED** [review]
Thanks for doing this! I had started it, but was lumping it in with #3911, which is huge and it definitely makes sense to separate this so we get it in the next release.

My comments are all minor, relating to variable and file naming, but mostly not optional. I think there may be some examples where pandas may issue a warning about writing to a view being unreliable, but I don't think any of these examples touch that. I will prioritize checking the other examples and fixing them as needed, to try to get them in for the next release.

File: examples/add_shapes_with_text.py (hunk 1 above)
Comment (jni) [reply examples/add_shapes_with_text.py]: :+1:

File: examples/nD_points_with_properties.py (hunk 2 above)
Comment (jni) [reply examples/nD_points_with_properties.py:22]: oh yeah, missed that, :+1:

**Comment by jni** [comment]
Thanks for the review @andy-sweet! Just pushed up commits implementing all your suggestions.

**Review by andy-sweet — APPROVED** [review]

**Comment by andy-sweet** [comment]
Thanks for the updates - looks good to me now!

**Comment by andy-sweet** [comment]
> The only example I haven't changed is bbox_annotator. That uses current_properties and default_properties and I didn't know off the top of my head how to port those to the new features API. I'll let @andy-sweet deal with that one in a separate PR.

As for this example, `property_choices` is easy enough to deal with, though requires explicit usage of `pandas.CategoricalDType` (or an equivalent categorical dtype-like), but I think that's fine.

`current_properties` is harder to replace because it absorbs quite a lot of behavior. For some related conversation, see #4064. The simplest solution is to re-open that issue and effectively make `feature_defaults` behave the same way as `current_properties` does now. I'm a little hesitant to do that because there's an opportunity to simplify that behavior by removing at least some of the selection-dependent behavior.

@jni if you can make tomorrow's architecture working group meeting, that seems like a good place/time to discuss our options in more detail.

**Comment by jni** [comment]
> there's an opportunity to simplify that behavior by removing at least some of the selection-dependent behavior.

Yup, let's simplify, I just wasn't quite sure how to do it! :sweat_smile: 

> @jni if you can make tomorrow's architecture working group meeting, that seems like a good place/time to discuss our options in more detail.

Yup I'll be there!

**Pull request merged**